packages/gatsby-image/src/index.js

```diff
@@ -415,7 +419,11 @@ class Image extends React.Component {
 
     if (fluid) {
       const imageVariants = fluid
-      const image = imageVariants[0]
+      const image = { ...imageVariants[0] }
+
+      if (hasSaveDataEnabled()) {
+        delete image.srcSet
+      }
```

Comment:
Should we instead use `hasSaveDataEnabled` check in place where we generate responsive sizes elements? Creating copy, deleting field and then checking for field seems pretty hacky?

Reply:
Unfortunately, we cannot. Those information are the result of GraphQL queries using gatsby-plugin-sharp. They're then passed as props to the Image component.
It creates a copy in order to keep the original object as is.

Reply:
Sure, but we don't need to delete `srcSet` field - we can just not use it if `hasSaveDataEnabled()` is true, right?